src/Fetcher/SymfonyFetcher.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Starfleet Project.
+ *
+ * (c) Starfleet <msantostefano@jolicode.com>
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace App\Fetcher;
+
+use App\Entity\Conference;
+use App\Entity\Continent;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class SymfonyFetcher implements FetcherInterface
+{
+    private const SOURCE = 'Symfony';
+    private const SYMFONY_SOURCE_URL = 'https://live.symfony.com/api/conference/all.json';
+
+    private LocationGuesser $locationGuesser;
+    private HttpClientInterface $client;
+    private LoggerInterface $logger;
+
+    public function __construct(LocationGuesser $locationGuesser, ?HttpClientInterface $client = null, ?LoggerInterface $logger = null)
+    {
+        $this->locationGuesser = $locationGuesser;
+        $this->client = $client ?: HttpClient::create();
+        $this->logger = $logger ?: new NullLogger();
+    }
+
+    public function fetch(array $configuration = []): \Generator
+    {
+        foreach ($this->querySymfony() as $conference) {
+            if (!\array_key_exists('ends_at', $conference) || !$conference['ends_at']['date']) {
+                continue;
+            }
+
+            $endDate = new \DateTime($conference['ends_at']['date']);
+
+            if (new \DateTime('now') < $endDate) {
+                yield $this->denormalizeConference($conference);
+            }
+        }
+    }
+
+    public function configureForm(FormBuilderInterface $formBuilder)
+    {
+    }
+
+    /** @param array<mixed> $rawConference */
+    private function denormalizeConference(array $rawConference): ?Conference
+    {
+        if (!$rawConference['is_online']) {
+            $city = $rawConference['city'];
+            $continent = $this->locationGuesser->getContinent($city);
+
+            if (!$continent->getEnabled() || !$continent instanceof Continent) {
+                return null;
+            }
+        }
+
+        $startDate = null;
+
+        if ($rawConference['starts_at']) {
+            $startDate = new \DateTimeImmutable($rawConference['starts_at']['date']);
+        }
+
+        // In case of invalid startDate, we skip the conference. It will be handled again later.
```

Comment:
If the date is invalid, it will generate an exception, isn't?
So it looks like this is dead code (so you can remove it)